### PR TITLE
[hotfix][docs] fix broken links

### PR DESCRIPTION
**Pull request by sjwiesman**
## What is the purpose of the change

Update broken links in the documentation introduced by FLINK-12639

### CODE REVIEW

**Comment by sjwiesman** [comment]
cc @knaufk

File: docs/dev/libs/state_processor_api.zh.md [hunk 1]
```diff
@@ -0,0 +1,330 @@
+---
```
Comment (knaufk) [inline docs/dev/libs/state_processor_api.zh.md]: I think, you checked this in by accident.

File: docs/redirects/filesystems.md [hunk 2]
```diff
@@ -1,8 +1,8 @@
 ---
 title: "File Systems"
 layout: redirect
-redirect: /ops/filesystems.html
-permalink: /ops/filesystems/index.html
+redirect: /ops/filesystems/index.html
```
Comment (knaufk) [inline docs/redirects/filesystems.md]: The filesystem link changes are unrelated to the original issue introducing the other problems. I don't see a problem in fixing these in the same commit, but please double-check.

**Comment by sjwiesman** [comment]
@knaufk I removed the state processor file. The filesystem one is a broken link I just happened to stumble upon but it is valid and needs to be fixed.

**Comment by sjwiesman** [comment]
@rmetzger Could you please merge?

**Comment by StephanEwen** [comment]
Thanks for fixing this. Merging...
